config/utils/isOutsideClick.ts

```diff
@@ -0,0 +1,17 @@
+type IsOutsideClickArgs = {
+  el: Element,
+  clickedEl: Element,
+  exclude?: Array<String>,
+}
+
+const isOutsideClick = ({ el, clickedEl, exclude = [] }: IsOutsideClickArgs) => {
+  const isWithin = el === clickedEl || el?.contains(clickedEl);
+  const clickOnExcluded = exclude.reduce(
+    (prev, selector) => Boolean(prev || clickedEl.closest(selector as keyof HTMLElementTagNameMap)),
+    false
+  );
+
+  return !(isWithin || clickOnExcluded);
+}
+
+export default isOutsideClick;
```

Comment:
@calvinhlee23 / @lightninglu10 just FYI - Created this helper function since we do this operation quite often

Reply:
Can you have a callback as an arg, `onClickOutside` and make this into a ueffect so we don't have to redefine it at the component level?